Android: couldn't import projects

### Expected Behavior

Can import projects

### Actual Behavior

When open project with the app, nothing happens. You can see from Logcat that the app lacks permission.
```
2021-10-08 21:55:48.860 28243-28243/org.scratchjr.android W/System.err: java.io.FileNotFoundException: /storage/emulated/0/Download/project13.sjr: open failed: EACCES (Permission denied)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at libcore.io.IoBridge.open(IoBridge.java:456)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at java.io.FileInputStream.<init>(FileInputStream.java:76)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at java.io.FileInputStream.<init>(FileInputStream.java:103)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at android.content.ContentResolver.openInputStream(ContentResolver.java:644)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at org.scratchjr.android.IOManager.receiveProject(IOManager.java:287)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at org.scratchjr.android.ScratchJrActivity$4.run(ScratchJrActivity.java:341)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at android.app.Activity.runOnUiThread(Activity.java:5309)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at org.scratchjr.android.ScratchJrActivity.importProject(ScratchJrActivity.java:337)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at org.scratchjr.android.ScratchJrActivity.receiveProject(ScratchJrActivity.java:320)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at org.scratchjr.android.ScratchJrActivity.onNewIntent(ScratchJrActivity.java:311)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1227)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1239)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at android.app.ActivityThread.deliverNewIntents(ActivityThread.java:2623)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at android.app.ActivityThread.performNewIntents(ActivityThread.java:2635)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at android.app.ActivityThread.handleNewIntent(ActivityThread.java:2644)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at android.app.ActivityThread.access$1600(ActivityThread.java:160)
2021-10-08 21:55:48.877 28243-28243/org.scratchjr.android W/System.err:     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1393)
2021-10-08 21:55:48.878 28243-28243/org.scratchjr.android W/System.err:     at android.os.Handler.dispatchMessage(Handler.java:102)
2021-10-08 21:55:48.878 28243-28243/org.scratchjr.android W/System.err:     at android.os.Looper.loop(Looper.java:135)
2021-10-08 21:55:48.878 28243-28243/org.scratchjr.android W/System.err:     at android.app.ActivityThread.main(ActivityThread.java:5604)
2021-10-08 21:55:48.878 28243-28243/org.scratchjr.android W/System.err:     at java.lang.reflect.Method.invoke(Native Method)
2021-10-08 21:55:48.878 28243-28243/org.scratchjr.android W/System.err:     at java.lang.reflect.Method.invoke(Method.java:372)
2021-10-08 21:55:48.878 28243-28243/org.scratchjr.android W/System.err:     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:984)
2021-10-08 21:55:48.878 28243-28243/org.scratchjr.android W/System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
2021-10-08 21:55:48.878 28243-28243/org.scratchjr.android W/System.err: Caused by: android.system.ErrnoException: open failed: EACCES (Permission denied)
2021-10-08 21:55:48.879 28243-28243/org.scratchjr.android W/System.err:     at libcore.io.Posix.open(Native Method)
2021-10-08 21:55:48.879 28243-28243/org.scratchjr.android W/System.err:     at libcore.io.BlockGuardOs.open(BlockGuardOs.java:186)
2021-10-08 21:55:48.879 28243-28243/org.scratchjr.android W/System.err:     at libcore.io.IoBridge.open(IoBridge.java:442)
2021-10-08 21:55:48.879 28243-28243/org.scratchjr.android W/System.err: 	... 23 more
```

### Steps to Reproduce

1. download a project from an email or browser
2. open the file and select ScratchJr
3. no new project appears in the Lobby.

### Device and Operating System

Kindle Fire HD 8 (Android 5.1)

### Note

After I add the line `<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>` to `AndroidManifest.xml`, the import function works fine. So we might need to rollback this PR https://github.com/LLK/scratchjr/pull/486 .

--- a/android/ScratchJr/app/src/main/AndroidManifest.xml
+++ b/android/ScratchJr/app/src/main/AndroidManifest.xml
@@ -6,6 +6,7 @@
     <uses-permission android:name="android.permission.INTERNET"/>
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
     <uses-permission android:name="android.permission.RECORD_AUDIO"/>
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
 
     <uses-feature android:name="android.hardware.camera" android:required="false" />
     <uses-feature android:name="android.hardware.microphone" android:required="false" />

--- a/android/ScratchJr/app/src/main/java/org/scratchjr/android/ScratchJrActivity.java
+++ b/android/ScratchJr/app/src/main/java/org/scratchjr/android/ScratchJrActivity.java
@@ -183,9 +183,11 @@ public class ScratchJrActivity
     public void requestPermissions() {
         cameraPermissionResult = ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA);
         micPermissionResult = ContextCompat.checkSelfPermission(this, Manifest.permission.RECORD_AUDIO);
+        readExtPermissionResult = ContextCompat.checkSelfPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE);
 
         if (cameraPermissionResult == PackageManager.PERMISSION_GRANTED
-            && micPermissionResult == PackageManager.PERMISSION_GRANTED) {
+            && micPermissionResult == PackageManager.PERMISSION_GRANTED
+            && readExtPermissionResult == PackageManager.PERMISSION_GRANTED) {
             return;
         }
 
@@ -195,6 +197,9 @@ public class ScratchJrActivity
         }
         if (micPermissionResult != PackageManager.PERMISSION_GRANTED) {
             tmp.add(Manifest.permission.RECORD_AUDIO);
+        }
+        if (readExtPermissionResult != PackageManager.PERMISSION_GRANTED) {
+            tmp.add(Manifest.permission.READ_EXTERNAL_STORAGE);
         }
         Object[] tmpArray = tmp.toArray();
         String[] desiredPermissions = Arrays.copyOf(tmpArray, tmpArray.length, String[].class);
@@ -215,6 +220,9 @@ public class ScratchJrActivity
                 }
                 if (permission.equals(Manifest.permission.RECORD_AUDIO)) {
                     micPermissionResult = grantResults[permissionId];
+                }
+                if (permission.equals(Manifest.permission.READ_EXTERNAL_STORAGE)) {
+                    readExtPermissionResult = grantResults[permissionId];
                 }
                 permissionId++;
             }

